common/src/java/org/apache/hadoop/hive/common/jsonexplain/Vertex.java

```diff
@@ -308,6 +309,16 @@ public void setType(String type) {
     this.edgeType = this.parser.mapEdgeType(type);
   }
 
+  @Override
+  public boolean equals(Object o) {
+    return  super.equals(o);
```

Comment:
interesting choices were made when this class was created...
the most important fields seems to be `String name, JSONObject vertexObject, Stage stage`
passing the `parser` in the constructor is an interesting idea...

although this `equals` resorts to identity comparision - I don't see that so out of scope for this class...

Reply:
You are right comparison could be implemented as part of the class (added that part).

Regarding the parser I agree it should not be a field, only given as parameter where needed -- there are more places where we follow this weird implementation and should probably be addressed all together: https://github.com/apache/hive/blob/ba0217ff17501fb849d8999e808d37579db7b4f1/common/src/java/org/apache/hadoop/hive/common/jsonexplain/Stage.java#L52